src/jit/lower.cpp

```diff
@@ -1280,11 +1280,16 @@ GenTreePtr Lowering::NewPutArg(GenTreeCall* call, GenTreePtr arg, fgArgTabEntryP
         // pair copying using XMM registers or rep mov instructions.
         if (fp->isStruct)
         {
-            assert(arg->OperGet() == GT_LDOBJ);
-            
+            unsigned numRefs = 0;
             BYTE* gcLayout = new (comp, CMK_Codegen) BYTE[fp->numSlots];
-
-            unsigned numRefs = comp->info.compCompHnd->getClassGClayout(arg->gtLdObj.gtClass, gcLayout);
+            if (arg->gtOper != GT_LDOBJ)
+            {
+                assert(varTypeIsSIMD(arg));
+            }
+            else
+            {
+                numRefs = comp->info.compCompHnd->getClassGClayout(arg->gtLdObj.gtClass, gcLayout);
```

Comment:
Is it valid to assert here that in else part 'arg' is not a SIMD type?
assert(!varTypeIsSIMD(arg))

A comment here will be useful why we don't expect a LDOBJ on a SIMD type.